agent/engine/docker_task_engine.go

```diff
@@ -625,7 +647,21 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
```

Comment:
I think I previously discussed making this a separate container state with @aaithal. Given the timeline of my ability to work on this project, I do not think I can feasibly make this change right now.